### PR TITLE
Add support for multiple created_resources

### PR DESCRIPTION
**Pull request by bmbouter**
When creating objects that need cleanup later, you may end up with
multiple created_resource objects in the task. This adds support for
that situation, and it will find the right type instead of raising an
Exception.

[noissue]

### CODE REVIEW

File: pulp_smash/pulp3/pytest_plugin/__init__.py [hunk 1]
```diff
@@ -500,16 +500,17 @@ def _gen_object_with_cleanup(api_client, *args, **kwargs):
...
-            new_obj = api_client.read(task_data.created_resources[0])
-            add_to_cleanup(api_client, new_obj.pulp_href)
+            for created_resource in task_data.created_resources:
+                try:
+                    new_obj = api_client.read(created_resource)
+                except Exception:
+                    pass  # This isn't the right created_resource for this api_client
+                else:
+                    add_to_cleanup(api_client, new_obj.pulp_href)
+                    return new_obj
```
Comment (dkliban) [inline pulp_smash/pulp3/pytest_plugin/__init__.py:510]: This will return after processing one created resource. Is that intentional?

Comment (bmbouter) [reply pulp_smash/pulp3/pytest_plugin/__init__.py:510]: It is, but there may be cases where we need more later? The issue with returning N is that the interface right now is only setup to return one object, not N, and it's already in broad use.

Comment (dkliban) [reply pulp_smash/pulp3/pytest_plugin/__init__.py:510]: In what use case do you expect it to return 2 resources? Uploading a file to a repository? new content and new repo version?

Comment (bmbouter) [reply pulp_smash/pulp3/pytest_plugin/__init__.py:510]: Yeah it's exactly that case, when creating a new file content unit and also specifying a repo at creation time it also creates the new repo version. That's why I need this actually.

**Review by dkliban — APPROVED** [review]

**Pull request merged**